base_user_role/views/user.xml

```diff
@@ -12,7 +12,11 @@
                 <page string="Roles">
                     <field name="role_line_ids" nolabel="1">
                         <tree editable="bottom" decoration-muted="not is_enabled">
-                            <field name="role_id" />
+                            <field name="user_roles_already_picked" invisible="1" />
+                            <field
+                                name="role_id"
+                                domain="[('id', 'not in', user_roles_already_picked)]"
```

Comment:
you can make it simplier.
There is already a computed field on the res.user see here https://github.com/OCA/server-backend/blob/14.0/base_user_role/models/user.py#L15

You can simply add this field in the view (in invisible) and then use the domain 
 domain="[('id', 'not in', parent.role_ids)]"

Reply:
changed + functional tested